apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1513,12 +1513,22 @@ def _update_dataframe(
         # schema-creation logic.
         atype = arrow_schema.field(add_key).type
         dtype = tiledb_type_from_arrow_type(atype)
+
+        enum_label = None
+        if pa.types.is_dictionary(arrow_table.schema.field(add_key).type):
+            enum_label = add_key
+            ordered = atype.ordered
+            dt = cast(pd.CategoricalDtype, new_data[add_key].dtype)
+            values = dt.categories
+            se.add_enumeration(tiledb.Enumeration(add_key, ordered, list(values)))
```

Comment:
Passing the things to `tiledb.Enumeration` as kwargs might make this easier to follow (particularly the `ordered` boolean parameter), and reduce the desire to give these temporary names.